### PR TITLE
Threshold and order of recent view signal by relevance (view count)

### PR DESCRIPTION
**Pull request by cbart**
This pull request:

- [x] Fixes a bug in `recentViewSignalStore.List`. When `Path:""` the store would return records for all paths, but there was no way to query repo root path - which is the more useful use case. Now we return repo root entries for `Path: ""`
- [x] Introduces ordering based on view count.
- [x] Introduces absolute threshold (that is - only show if more than N views)

## Test plan

Update existing database tests.

### CODE REVIEW

File: internal/database/recent_view_signal_test.go [hunk 1]
```diff
@@ -446,86 +446,107 @@ func TestRecentViewSignalStore_List(t *testing.T) {
 
 	t.Parallel()
 	logger := logtest.Scoped(t)
-	db := NewDB(logger, dbtest.NewDB(logger, t))
+	d := NewDB(logger, dbtest.NewDB(logger, t))
 	ctx := context.Background()
 
 	// Creating a user.
```
Comment (sashaostrikov) [inline internal/database/recent_view_signal_test.go:452]:
```suggestion
	// Creating 2 users.
```

File: internal/database/recent_view_signal_test.go [hunk 2]
```diff
@@ -446,86 +446,107 @@ func TestRecentViewSignalStore_List(t *testing.T) {
...
 
 	// Creating a repo.
-	err = db.Repos().Create(ctx, &types.Repo{ID: 1, Name: "github.com/sourcegraph/sourcegraph"})
+	var repoID api.RepoID = 1
+	err = d.Repos().Create(ctx, &types.Repo{ID: repoID, Name: "github.com/sourcegraph/sourcegraph"})
 	require.NoError(t, err)
 
 	// Creating some paths.
-	_, err = db.QueryContext(ctx, "INSERT INTO repo_paths (repo_id, absolute_path, parent_id) VALUES (1, '', NULL), (1, 'src', 1), (1, 'src/abc', 2), (1, 'src/cde', 2)")
+	// TODO DROP
```
Comment (sashaostrikov) [inline internal/database/recent_view_signal_test.go:464]: don't forget to DROP :)

**Review by sashaostrikov — APPROVED** [review]
Thanks for the fix!

File: internal/database/recent_view_signal_test.go (hunk 2 above)
Comment (cbart) [reply internal/database/recent_view_signal_test.go:464]: actually drop the todo, the API requires path IDs here.

**Pull request merged**